storage/innobase/xtrabackup/src/ddl_tracker.h

```diff
@@ -0,0 +1,194 @@
+/******************************************************
+Copyright (c) 2023 Percona LLC and/or its affiliates.
+
+DDL Tracker for XtraBackup.
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; version 2 of the License.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
+
+*******************************************************/
+#ifndef DDL_TRACKER_H
+#define DDL_TRACKER_H
+#include <univ.i>
+#include "log0types.h"
+#include "mtr0types.h"
+typedef std::unordered_map<space_id_t, std::string> space_id_to_name_t;
+typedef std::unordered_map<std::string, space_id_t> name_to_space_id_t;
+typedef std::unordered_map<space_id_t, std::string> meta_map_t;
+using filevec = std::vector<std::pair<std::string, space_id_t>>;
+struct datadir_entry_t;
+extern meta_map_t meta_map;
+
+inline const std::string EXT_DEL = ".del";
+inline const std::string EXT_REN = ".ren";
+inline const std::string EXT_NEW = ".new";
+inline const std::string EXT_IBD = ".ibd";
+inline const std::string EXT_IBU = ".ibu";
+inline const std::string EXT_META = ".meta";
+inline const std::string EXT_NEW_META = ".new.meta";
+inline const std::string EXT_DELTA = ".delta";
+inline const std::string EXT_NEW_DELTA = ".new.delta";
+
+class ddl_tracker_t {
+  /** List of all tables in the backup */
+  space_id_to_name_t tables_in_backup;
+  /** Tablspaces with their ID and name, as they were copied to backup.*/
+  space_id_to_name_t new_tables;
+  name_to_space_id_t before_lock_undo;
+  name_to_space_id_t after_lock_undo;
+  /** Tablespaces involved in encryption or bulk index load.*/
+  std::unordered_set<space_id_t> recopy_tables;
+  /** Drop operations found in redo log. */
+  space_id_to_name_t drops;
+  /* For DDL operation found in redo log,  */
+  std::unordered_map<space_id_t, std::pair<std::string, std::string>> renames;
+  /** Tables that have been deleted between discovery and file open */
+  std::unordered_set<std::string> missing_tables;
```

Comment:
instead of missing_tables, please use missing_afer_discovery.

Reply:
AB